app/Providers.tsx

```diff
@@ -0,0 +1,32 @@
+'use client'
+
+import { makeElectricContext } from 'electric-sql/react'
+import { useEffect, useState } from 'react'
+
+import { Electric } from '../src/generated/client'
+import { initElectric } from './initElectric'
+
+const { ElectricProvider, useElectric } = makeElectricContext<Electric>()
+
+interface Props {
+  children: React.ReactNode
+}
+
+export function Providers({ children }: Props) {
+  const [electric, setElectric] = useState<Electric | null>(null)
+  useEffect(() => {
+    async function f() {
+      const { electric } = await initElectric()
+      setElectric(electric)
+    }
+    f()
+  }, [])
+
+  if (!electric) {
+    return <div>{children}</div>
```

Comment:
This makes the app progressive BUT the user of useElectric must now check if it is instantiated or not.